web-marketplace/src/components/organisms/LocationForm/LocationForm.tsx

```diff
@@ -92,12 +94,27 @@ const LocationForm: React.FC<LocationFormProps> = ({
           error={!!errors['schema:location']}
           helperText={errors['schema:location']?.message}
           value={location}
-          handleChange={value =>
+          handleChange={value => {
             setValue('schema:location', value, {
               shouldDirty: true,
               shouldTouch: true,
-            })
-          }
+            });
+            // If a jurisdiction is provided (i.e. the project exists on chain),
+            // we require the location input to match the jurisdiction.
+            if (jurisdiction) {
+              let valid: boolean;
+              if (isGeocodingFeature(location)) {
+                valid = location.properties.short_code === jurisdiction;
+              } else {
+                valid = location === jurisdiction;
```

Comment:
I believe we shouldn't show any error message before the user selects a location from the search results.